src/Types.hs

```diff
@@ -285,6 +301,17 @@ areUnifiable FuncTy{} _ = False
 areUnifiable a b | a == b    = True
           | otherwise = False
 
+-- Checks whether or not the kindedness of types match
+-- Kinds are polymorphic constructors such as (f a)
+-- Note that this disagrees with the notion of unifiablitity in areUnifiable
+checkKinds :: Ty -> Ty -> Bool
+-- Base < Higher
+checkKinds (FuncTy argTysA retTyA _) (FuncTy argTysB retTyB _) =
+  let argKinds = zipWith (<=) (map tyToKind argTysA) (map tyToKind argTysB)
+      retKinds = tyToKind retTyA <= tyToKind retTyB
+  in all (== True) (retKinds : argKinds)
+checkKinds t t' = tyToKind t == tyToKind t'
```

Comment:
Maybe this should match the logic above? Instead of equals: `t <= t`. As of right now though, this function is only called against `Fn` types.

Reply:
Actually, if we define it this way we can call the function recursively instead of mapping `tyToKind`

Reply:
I went ahead and made this change.